src/state/chartDataSlice.ts

```diff
@@ -22,6 +22,11 @@ export type AppliedChartData = ReadonlyArray<{ value: unknown }>;
 
 export type ChartDataState = {
   chartData: ChartData | undefined;
+  /**
```

Comment:
bleh this could be better. Not sure if we even want it for all the charts?.... idk

Reply:
This is derived data, we should not store it explicitly unless necessary. Other charts use a selector instead.
The data goes through a lifecycle:

1. When it's the first, raw version, it's `data`
2. After zoom (dataStartIndex + dataEndIndex) it becomes `displayedData`
3. After processing by charts (applied dataKey plus whatnot) it becomes `appliedValues`

Can we stick to this pipeline?

If it's necessary to save some of these to the store then we can do that but usually it's not.

If you want to inspect the results of a selector, use unit test or the inspector in storybook

Reply:
Why not store derived data? Because it will get out of sync. We're seeing that in the current generator: there are so many caches and they are all inconsistent with each other.